pkg/reconciler/reconciler.go

```diff
@@ -354,6 +358,18 @@ func WithPreHook(h hook.PreHook) Option {
 	}
 }
 
+// WithPreExtension is an Option that configures the reconciler to run the given
+// extension before performing any reconciliation steps (including values translation).
+// The extension will be invoked with the raw object state; meaning it is responsible
+// for determining if a deletion needs to be performed by checking the deletionTimestamp
+// field.
```

Comment:
On first look I got the impression that this means that the mechanism for checking `deletionTimestamp` is now the extensions. How about this:

```suggestion
// The extension will be invoked with the raw object state; meaning it needs to be careful
// to check for existence of the deletionTimestamp field.
```
If it LGTY please also apply this to `WithPostExtension`.